contracts/contracts/EventManager.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+contract EventManager {
+    struct Event {
+        uint256 eventId;
+        uint16 eventType;
+        bytes data;
+    }
+
+    mapping(uint => Event[]) private _eventsByType;
```

Comment:
did you consider using a map of maps instead of a map of vectors? 

currently, the complexity is as follows:
- `insertEvent`: `O(1)`
- `eraseEvents`: `O(n)` where `n` is the number of events of the given type
- `pollEvents`: `O(1)`

`insertEvent` and `eraseEvents` are the most intense operations as `pollEvents` is infrequent compared to them.

if you were to use a map of maps, then `insertEvent` and `eraseEvents` would become `O(1)` while `pollEvents` would be `O(n)`, which is more suitable as i see it.
```
mapping(uint256 => Event) private _eventsByTypeMapping;
mapping(uint16 => uint256[]) private _eventTypeToEventIds;
```